src/core_editor/line_buffer.rs

```diff
@@ -902,13 +922,25 @@ mod test {
     }
 
     #[rstest]
-    #[case("This is a te", 4)]
-    #[case("This is a test", 4)]
-    #[case("This      is a test", 4)]
-    fn word_count_works(#[case] input: &str, #[case] expected_count: usize) {
+    #[case("This is a test", 3)]
+    #[case("This is a test  ", 3)]
+    #[case("  This is a test", 4)]
+    #[case("This  \n    is a test", 4)]
+    fn word_start_works(#[case] input: &str, #[case] expected_count: usize) {
```

Comment:
Just to reduce possible confusion: 
```suggestion
    fn word_starts_works(#[case] input: &str, #[case] expected_count: usize) {
```